File: Solutions/0038- Count and Say/Solution.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return "1"
        
        number = self.countAndSay(n-1)
        index = 0
        answer = ""

        while (index < len(number)):
            temp = number[index]
            count = 0
            while (index < len(number) and number[index] == temp):
                index += 1
                count += 1
            answer += str(count)
            answer += str(temp)
        
        return answer
        

File: Solutions/0038- Count and Say/Solution.java
class Solution {
    public String countAndSay(int n) {
        if (n == 1)
            return "1";

        StringBuilder number = new StringBuilder(countAndSay(n-1));
        int index = 0;
        StringBuilder answer = new StringBuilder();

        while (index < number.length()) {
            char temp = number.charAt(index);
            int count = 0;
            while (index < number.length() && number.charAt(index) == temp) {
                index++;
                count++;
            }
            answer.append(count);
            answer.append(temp);
        }

        return answer.toString();

    }

}

